engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeConversionNode.java

```diff
@@ -102,11 +109,15 @@ Object doConvertFrom(
       Object that,
       Object[] arguments,
       @Shared("typesLib") @CachedLibrary(limit = "10") TypesLibrary dispatch,
-      @Shared("conversionResolverNode") @Cached ConversionResolverNode conversionResolverNode) {
-    Function function =
-        conversionResolverNode.expectNonNull(
-            that, extractConstructor(self), dispatch.getType(that), conversion);
-    return invokeFunctionNode.execute(function, frame, state, arguments);
+      @Shared("conversionResolverNode") @Cached ConversionResolverNode resolveNode) {
+    var thatType = dispatch.getType(that);
+    if (thatType == self) {
+      return that;
```

Comment:
This change is needed to support the 
````
h = Hello.Say "x"
h.to Hello
```
[idempotent conversion](https://github.com/enso-org/enso/pull/7704/files#diff-830a1dac9433e5ae1a2e87c7da0e83f7c9c90b0999f94422e403494ff166721bR136). Previously nobody would call `Hello.from h`, but with `.to` method this is [going to be common](https://github.com/enso-org/enso/pull/7704/files#diff-97b17337e908ef4d3eda8ce7e0edd6d87dc0a8e044f005f3b97959720137659bR14) as one just requets `h.to Hello` conversion without checking the actual type of `h`.